awgreene/oria-operator
# Added initial test cases

This test suit follows the below steps.

## Test Case 1
1. Create a `test` namespace
2. Create `ScopeTemplate` as specified in the test cases with a single cluster role.
3. It verifies that the ScopeTemplate is created or not
4. Create `ScopeInstance` as specified in the test cases with a single namespace added as `test`
5.  It verifies that the ScopeInstace is created or not
6. Then it verifies whether the `ClusterRole` is created or not and matches each and every field like owner reference, subjects, rules, etc.
7. Then it verifies that the RoleBinding is created or not in the `test` namespace and it verifies every field like `owner reference, subjects, roleref` etc
8. Delete ScopeInstance.

## Test Case 2
1. Create a `testanotherns` namespace
2. Create `ScopeInstance` as specified in the test cases with namespaces added as `test, testanotherns`
3.  It verifies that the ScopeInstace is created or not
4. Then it verifies that the RoleBinding is created or not in the `test` namespace and it verifies every field like `owner reference, subjects, roleref` etc
5. Then it verifies if RoleBinding is created or not in the `testanotherns` namespace and it verifies every field like `owner reference, subjects, roleref` etc
6. Delete ScopeInstance

## Test Case 3
1. Remove one of the namespaces from the above two from Scope Instance.
2. Verify the behavior, it should delete the respective role binding.

## Test Case 4
1. Remove all the namespaces from scope Instance and update it.
2. It should delete all the role bindings and must create cluster role bindings.
